### PR TITLE
E2E: terraform provisioner upgrades

### PR DESCRIPTION
**Pull request by tgross**
While working on infrastructure for testing the UI in E2E, we needed
to upgrade the certificate provider. Performing a provider upgrade via
the TF `init -upgrade` brought in updates for the file and AWS
providers as well. These updates include deprecating the use of
`sensitive_content` fields, removing CA algorithm parameters that can
be inferred from keys, and removing the requirement to manually
specify AWS assume role parameters in the provider config if they're
available in the calling environment's AWS config file (as they are
via doormat or our E2E environment).

I've pulled this out to its own PR so as not to muddy the upcoming 
UI E2E infra PR.

### CODE REVIEW

File: e2e/terraform/hcp_consul.tf [hunk 1]
```diff
@@ -30,36 +30,36 @@ data "consul_acl_token_secret_id" "consul_agent_token" {
...
 }
 
-resource "local_file" "consul_config_file" {
-  sensitive_content = base64decode(data.hcp_consul_cluster.e2e_shared_consul.consul_config_file)
-  filename          = "uploads/shared/consul.d/consul_client.json"
-  file_permission   = "0744"
+resource "local_sensitive_file" "consul_config_file" {
+  content         = base64decode(data.hcp_consul_cluster.e2e_shared_consul.consul_config_file)
+  filename        = "uploads/shared/consul.d/consul_client.json"
+  file_permission = "0744"
```
Comment (shoenig) [inline e2e/terraform/hcp_consul.tf:50]:
```suggestion
  file_permission = "0644"
```

File: e2e/terraform/hcp_consul.tf [hunk 2]
```diff
@@ -30,36 +30,36 @@ data "consul_acl_token_secret_id" "consul_agent_token" {
...
 }
 
-resource "local_file" "consul_base_config_file" {
-  sensitive_content = templatefile("${path.root}/etc/consul.d/clients.json", {})
-  filename          = "uploads/shared/consul.d/consul_client_base.json"
-  file_permission   = "0744"
+resource "local_sensitive_file" "consul_base_config_file" {
+  content         = templatefile("${path.root}/etc/consul.d/clients.json", {})
+  filename        = "uploads/shared/consul.d/consul_client_base.json"
+  file_permission = "0744"
```
Comment (shoenig) [inline e2e/terraform/hcp_consul.tf:56]:
```suggestion
  file_permission = "0644"
```

File: e2e/terraform/hcp_consul.tf [hunk 3]
```diff
@@ -30,36 +30,36 @@ data "consul_acl_token_secret_id" "consul_agent_token" {
...
 }
 
-resource "local_file" "consul_systemd_unit_file" {
-  sensitive_content = templatefile("${path.root}/etc/consul.d/consul.service", {})
-  filename          = "uploads/shared/consul.d/consul.service"
-  file_permission   = "0744"
+resource "local_sensitive_file" "consul_systemd_unit_file" {
+  content         = templatefile("${path.root}/etc/consul.d/consul.service", {})
+  filename        = "uploads/shared/consul.d/consul.service"
+  file_permission = "0744"
```
Comment (shoenig) [inline e2e/terraform/hcp_consul.tf:62]:
```suggestion
  file_permission = "0644"
```

**Review by DerekStrickland — APPROVED** [review]
:shipit:

File: e2e/terraform/provision-nomad/install-linux.tf [hunk 4]
```diff
@@ -1,7 +1,7 @@
-resource "local_file" "nomad_systemd_unit_file" {
-  sensitive_content = templatefile("etc/nomad.d/nomad-${var.role}.service", {})
-  filename          = "${local.upload_dir}/nomad.d/nomad.service"
-  file_permission   = "0700"
+resource "local_sensitive_file" "nomad_systemd_unit_file" {
+  content         = templatefile("etc/nomad.d/nomad-${var.role}.service", {})
+  filename        = "${local.upload_dir}/nomad.d/nomad.service"
+  file_permission = "0700"
```
Comment (shoenig) [inline e2e/terraform/provision-nomad/install-linux.tf:4]:
```suggestion
  file_permission = "0600"
```

**Review by shoenig — APPROVED** [review]
May as well fix some file permissions; otherwise LGTM

**Pull request merged**